aten/src/ATen/native/cudnn/Conv_v8.cpp

```diff
@@ -76,101 +85,238 @@ void filterEngineConfigs(
+    return filtered_configs;
+  };
+  std::array<cudnn_frontend::GeneratorSource const, 2> sources = {heurgen_method, fallback_method};
+  return sources;
+}
+
+const auto get_fallback_method(const cudnn_frontend::OperationGraph &opGraph, const cudnnBackendDescriptorType_t& desc, const Tensor& x, const bool deterministic, const bool allow_tf32) {
+  // Method for engine config generator based on fallback list
+  auto fallback_method = [&](cudnn_frontend::OperationGraph &opGraph) -> cudnn_frontend::EngineConfigList {
+    auto fallback = cudnn_frontend::EngineFallbackListBuilder()
+                        .setOperationGraph(opGraph)
+                        .setOperation(desc)
+                        .build();
+    auto &fallback_list = fallback.getFallbackList();
+    cudnn_frontend::EngineConfigList filtered_configs;
+    filterEngineConfigs(fallback_list, filtered_configs, deterministic, allow_tf32, x.scalar_type());
+    return filtered_configs;
+  };
+  return fallback_method;
+}
+
+auto get_plans_from_find(const cudnnHandle_t handle, const cudnnBackendDescriptorType_t desc, const Tensor& x, const Tensor& y, const Tensor& w, const CacheKey& key, const IntArrayRef padding, const IntArrayRef stride, const IntArrayRef dilation, const bool deterministic, const bool allow_tf32) {
+  auto opGraph = build_opgraph(handle, desc, x, y, w, key, padding, stride, dilation);
+  void *data_ptrs[] = {x.data_ptr(), y.data_ptr(), w.data_ptr()};
+  int64_t uids[] = {'x', 'y', 'w'};
+
+  auto sources = get_generator_sources(opGraph, desc, x, deterministic, allow_tf32);
+  auto initial_predicate_function = [&](cudnn_frontend::ExecutionPlan const& plan) -> bool {
+    return false;
+  };
+  cudnn_frontend::EngineConfigGenerator generator(sources.size(), sources.data());
+  size_t max_workspace_size = 0u;
+  auto plans = generator.cudnnGetPlan(handle, std::move(opGraph), initial_predicate_function);
 
-  auto heuristics = cudnn_frontend::EngineHeuristicsBuilder()
-      .setOperationGraph(opGraph)
-      .setHeurMode(CUDNN_HEUR_MODE_INSTANT)
+  int device;
+  THCudaCheck(cudaGetDevice(&device));
+  size_t max_block_size = 0;
+  size_t tmp_bytes = 0;  // Only used for filling pointer parameters that aren't used later
+  c10::cuda::CUDACachingAllocator::cacheInfo(device, &tmp_bytes, &max_block_size);
+  cudnn_frontend::executionPlans_t valid_plans;
+
+  std::for_each(plans.begin(), plans.end(), [&max_workspace_size, &max_block_size, &valid_plans](cudnn_frontend::ExecutionPlan& plan) { 
+    size_t curr_workspace_size = plan.getWorkspaceSize();
+    if (curr_workspace_size <= max_block_size) {
+      if (curr_workspace_size > max_workspace_size && curr_workspace_size <= max_block_size) {
+        max_workspace_size = plan.getWorkspaceSize();
+      }
+      valid_plans.emplace_back(std::move(plan));
+    }
+  });
+  TORCH_CHECK_WITH(CUDAOutOfMemoryError, max_workspace_size < 1_TiB, "Not enough memory for workspace!");
```

Comment:
This is very verbose and confusing. Do you think we should file some feature requests to the cuDNN team to add logic like these into cuDNN frontend?